src/Categorize.hs

```diff
@@ -54,11 +55,11 @@ data Ctx = Ctx
         , cSubsts :: [Text]
         , cCurrentTime :: UTCTime
         , cTimeZone :: TimeZone
-        }
-  deriving (Show)
+        , letBindings :: Map String Cond
```

Comment:
This indicates that the condition will be evaluated upon every use. But why not `Map String Bool`, and evaluate the condition only once, upon assignment? Seems to be more efficient when the condition is used multiple times.

Reply:
Can we really skip evaluation?  What if we refer to `current window $title` in our condition binding?